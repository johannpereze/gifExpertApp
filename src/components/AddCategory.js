import { useState } from "react";
export const AddCategory = () => {
  const [inputValue, setInputValue] = useState("Hola Mundo");

  const handleInputChange = ()=>{
      
  }

  return (
    <>
      <input type="text" value={inputValue} onChange={console.log} />
    </>
  );
};
