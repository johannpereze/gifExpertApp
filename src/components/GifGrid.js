// import { useEffect, useState } from "react";
// import { getGifs } from "../helpers/getGifs";

import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";
import PropTypes from "prop-types"
require("dotenv").config();

export const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      <h3 className={"animate__animated animate__fadeIn"}>{category}</h3>
      {loading && <p>Cargando...</p>}
      <div className={"card-grid"}>
        {images.map((img) => {
          return <GifGridItem key={img.id} {...img} />;
        })}
      </div>
    </>
  );
};


GifGrid.prototypes ={
  category: PropTypes.string.isRequired
}