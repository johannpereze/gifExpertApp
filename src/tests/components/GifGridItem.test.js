import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe("Pruebas del GifGridItem component", () => {
    const title = "Esto es un título"
    const url = "https://localhost/algo.jpg"

    const wrapper = shallow(<GifGridItem title={title} url={url} />);

    test("debería renderizar correctamente el componente haciendo match con el snapshot", () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de tener un párrafo con el title', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title)
    })
    test('Debe de tener una iamgen con su URL', () => {
        const img = wrapper.find('img');
        expect(img.props().alt).toBe(title)
    })

    test('debe tener la clase animate__fadeIn', () => {
        const fadeInClass = "animate__fadeIn"

        const fadeInDiv = wrapper.find('div');

        expect(fadeInDiv.props().className).toContain(fadeInClass)
    })


});