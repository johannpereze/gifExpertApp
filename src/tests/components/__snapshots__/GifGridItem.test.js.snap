// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del GifGridItem component debería renderizar correctamente el componente haciendo match con el snapshot 1`] = `
<div
  className="card  animate__animated animate__fadeIn"
>
  <img
    alt="Esto es un título"
    src="https://localhost/algo.jpg"
  />
  <p>
    Esto es un título
  </p>
</div>
`;
